packages/forms-web-app/src/dynamic-forms/journey.js

```diff
@@ -206,14 +206,14 @@ class Journey {
 					reverse ? j-- : j++
 				) {
 					const question = currentSection.questions[j];
-					if (takeNextQuestion) {
+					if (takeNextQuestion && question.shouldDisplay(this.response)) {
 						return this.#buildQuestionUrl(
 							currentSection.segment,
 							question.url ? question.url : question.fieldName
 						);
 					}
 
-					if (question.fieldName === questionSegment) {
+					if (!takeNextQuestion && question.fieldName === questionSegment) {
```

Comment:
is this needed? I feel like it still takes the next question without this

Also now we'll have a consistent set of sections/questions it feels like it should be possible to simplify this next question logic, for another time though

Reply:
IIRC this didn't work without that when I tested it but I might be wrong. I'll experiment

Reply:
you were right it didn't need this, removed